### DISCUSSION TITLE
Performance measurement on aihwkit

### DISCUSSION THREAD

**Discussion by mingyuan-xiang** [IBM/aihwkit#441] [Q&A] [answered]
Hi,

I'm a new user of aihwkit and want to ask whether aihwkit can measure the latency and energy consumption, mainly for inference, on RRAM simulation? If it's supported, do you have any pointer to the examples which I can look at. If it's not supported, will it be difficult for me to add the performance measurement by myself?

Thanks,
Mingyuan

**Comment by maljoras** [comment, marked as answer]
Hi @mingyuan-xiang ,
currently the AIHWKIT only does accuracy simulation, thus functional verification. It does not estimate energy or latency for RRAMs. 

We have a utility tool ([analog_info](https://github.com/IBM/aihwkit/blob/master/src/aihwkit/utils/analog_info.py)) that summarizes the number of crossbars and the tile utilization for a given DNN. In principle, one could use that information to get a rough estimate on latency and energy. If only a rough estimate is required that might be a good base and it would be a great contribution if you have some expertise to that end. However, for more accuracy estimation of latency, a complete chip-design and a compiler would be needed (adding all kinds of storage operations and communications etc), which is in itself very difficult to add and would be beyond the scope of this project.

**Reply by mingyuan-xiang** [reply]
Hi @maljoras ,

Thanks for your reply. I'll try out the estimation and see what I can get.

Mingyuan
